src/SearchByTitleForm/SearchByTitleForm.js

```diff
@@ -1,34 +1,42 @@
 import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
 import '../Navbar/Navbar.css';
-import { withRouter } from 'react-router';
 
 class SearchByNameForm extends Component {
-    constructor(props) {
-        super(props);
+    constructor() {
+        super();
         this.state = {
             movieTitle: '',
+            movieID: '',
+            submit: false,
             error: ''
         }
     }
+
+    
     
     handelMovieTitle = (event) => {
-        event.preventDefault()
-        this.setState({movieTitle: event.target.value });
+        this.setState({ movieTitle: event.target.value });
     }
     
-    resetMovieTitle = () => {
+    resetForm = () => {
         this.setState({ movieTitle: '' });
     }
 
     handelSubmit = (event) => {
         event.preventDefault();
         const id = this.props.findMovieByTitle(this.state.movieTitle);
         if(id) {
-            this.props.history.push('/' + id);
-            this.resetMovieTitle();
+            this.setState({ ...this.state, movieID: id, submit: true })
         } else {
             this.setState({...this.state, error: 'Looks like we can\'t find this title.'})
-            this.resetMovieTitle();
+            this.resetForm();
+        }
+    }
+
+    componentDidUpdate() {
+        if(this.state.submit) {
+            this.setState({ submit: false, movieID: '', error: '', movieTitle: '' })
```

Comment:
Here is the logic that resets the form, this runs when the `this.state.submit` is true